server/src/test/java/org/opensearch/action/admin/cluster/remotestore/restore/RestoreRemoteStoreResponseTests.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.restore;
+
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.snapshots.RestoreInfo;
+import org.opensearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RestoreRemoteStoreResponseTests extends AbstractXContentTestCase<RestoreRemoteStoreResponse> {
+
+    @Override
+    protected RestoreRemoteStoreResponse createTestInstance() {
+        if (randomBoolean()) {
```

Comment:
I've always found these `randomBoolean()` usages to be odd in a test case like this. Why not just always test both permutations?

Reply:
Agree, let me add tests for both the values.

Reply:
Actually, this is the implementation of the abstract method from `AbstractXContentTestCase` class. Based on the javadocs, the implementation is doing what is expected.

https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/AbstractXContentTestCase.java#L238-L243

Reply:
That makes sense for computing random values within defined bounds for things like string or integer fields, or even a boolean field where for the purposes of serialization it shouldn't be important if the value is true or false. But in this case you have a bimodal class where it does different things based on whether `restoreInfo` is null or not. I think it would be better to ensure both modes are tested on every unit test run. Not really a big deal if you'd have to fight against the framework to do it, but just something that has always struck me as odd.